client/trino-client/src/main/java/io/trino/client/StatementStats.java

```diff
@@ -197,7 +199,36 @@ public OptionalDouble getProgressPercentage()
         if (!scheduled || totalSplits == 0) {
             return OptionalDouble.empty();
         }
-        return OptionalDouble.of(min(100, (completedSplits * 100.0) / totalSplits));
+
+        if (isStageScheduled(rootStage)) {
```

Comment:
For `INSERT` queries with fault tolerant execution enabled the root stage will be scheduled right away

Reply:
Then let pipelined execution and fault tolerant execution share the same code path? It should change nothing for pipelined execution progress bar